### ISSUE TITLE
Inferences are taking 8 seconds on the dev board

### ISSUE THREAD

**Issue by sairaghava7** [google-coral/edgetpu#646]
I'm working on age and emotion recognition using a coral board. The inferences on my development machine are about 120 ms but whereas the inferences on the board take 6-8 seconds. I'm using quantized models and compiled them for edge tpu. Can anyone take a look at my code and provide suggestions? 

from tensorflow.keras.utils import img_to_array
from keras.preprocessing import image
import cv2
import numpy as np

face_cascade=cv2.CascadeClassifier('haarcascade_frontalface_default.xml')
## Load the TFLite model and allocate tensors.

emotion_interpreter = tf.lite.Interpreter(model_path="emotion_detection.tflite")
emotion_interpreter.allocate_tensors()
age_interpreter = tf.lite.Interpreter(model_path="C:/Users/sethuri/Documents/AGS_FR_T1/coursera/age_modelv4.tflite")
age_interpreter.allocate_tensors()

gender_interpreter = tf.lite.Interpreter(model_path="gender_detection.tflite")
gender_interpreter.allocate_tensors()

## Get input and output tensors.
emotion_input_details = emotion_interpreter.get_input_details()
emotion_output_details = emotion_interpreter.get_output_details()

age_input_details = age_interpreter.get_input_details()
age_output_details = age_interpreter.get_output_details()

gender_input_details = gender_interpreter.get_input_details()
gender_output_details = gender_interpreter.get_output_details()

# Test the model on input data.
emotion_input_shape = emotion_input_details[0]['shape']
age_input_shape = age_input_details[0]['shape']
gender_input_shape = gender_input_details[0]['shape']

#frame = cv2.imread('obamas.jpg')
###########################

output_indexes  = [i for i in range(101)]
#output_indexes = np.array([i for i in range(0, 101)])
class_labels=['Angry', 'Fear', 'Happy','Neutral','Sad','Surprise']
gender_labels = ['Male', 'Female']

cap=cv2.VideoCapture(0)

while True:
    ret,frame=cap.read()
    labels=[]
    
    gray=cv2.cvtColor(frame,cv2.COLOR_BGR2GRAY)
    faces=face_cascade.detectMultiScale(gray,1.3,5)

    for (x,y,w,h) in faces:
        cv2.rectangle(frame,(x,y),(x+w,y+h),(255,0,0),2)
        roi_gray=gray[y:y+h,x:x+w]
        roi_gray=cv2.resize(roi_gray,(48,48),interpolation=cv2.INTER_AREA)

        #Get image ready for prediction
        roi=roi_gray.astype('float32')/255.0  #Scale
        #roi = cv2.imread(roi)
        roi = np.array(roi).reshape(-1,48,48,1)
        #roi=img_to_array(roi)
        #roi = np.array(roi_gray, dtype=np.float32)
        #roi = roi/255.
        #roi=np.expand_dims(roi,axis=0)  #Expand dims to get it ready for prediction (1, 48, 48, 1)
        
        emotion_interpreter.set_tensor(emotion_input_details[0]['index'], roi)
        emotion_interpreter.invoke()
        emotion_preds = emotion_interpreter.get_tensor(emotion_output_details[0]['index'])

        #preds=emotion_model.predict(roi)[0]  #Yields one hot encoded result for 7 classesqq
        emotion_label=class_labels[emotion_preds.argmax()]  #Find the label
        emotion_label_position=(x,y)
        cv2.putText(frame,emotion_label,emotion_label_position,cv2.FONT_HERSHEY_SIMPLEX,1,(0,255,0),2)

        #Gender
        roi_color = frame[y:y+h,x:x+w]
        roi_color = cv2.resize(roi_color,(224,224),interpolation=cv2.INTER_AREA)
        roi_color = roi_color.astype('float32')/255.0
        roi_color = np.array(roi_color).reshape(-1,224,224,3) #input shape is (1, 200,200,3)
        #roi_color = roi_color.astype(np.float32)
        
        age_interpreter.set_tensor(age_input_details[0]['index'], roi_color)
        age_interpreter.invoke()
        age_preds = age_interpreter.get_tensor(age_output_details[0]['index'])
        age_label = output_indexes[age_preds.argmax()]
        #age =(int((np.sum(age_preds * output_indexes, axis = 1))))
        
        #age = round(age_preds[0,0])
        age_label_position=(x+h, y+h)
        cv2.putText(frame,"Age="+str(age_label),age_label_position,cv2.FONT_HERSHEY_SIMPLEX,1,(0,255,0),2)
     
    cv2.imshow('Emotion Detector', frame)
    if cv2.waitKey(1) & 0xFF == ord('q'):  #Press q to exit
        break
cap.release()
cv2.destroyAllWindows()`

**Comment by hjonnala** [comment]
> age_interpreter = tf.lite.Interpreter(model_path="C:/Users/sethuri/Documents/AGS_FR_T1/coursera/age_modelv4.tflite")
> gender_interpreter = tf.lite.Interpreter(model_path="gender_detection.tflite")

Hi @sairaghava7 Seems to be you are running the tflite models(which are not compiled for EdgeTPU) on Dev Board. The performance difference is due to your desktop CPU and Dev Board CPU. 

Feel free to try some pycoral examples located at: https://github.com/google-coral/pycoral/tree/master/examples
